ui/src/app/shared/genericComponents/modal/modal-value-line/modal-value-line.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, Input } from "@angular/core";
+import { ChannelAddress, CurrentData } from "src/app/shared/shared";
+
+import { AbstractModalLine } from "../abstract-modal-line";
+
+@Component({
+  selector: 'oe-modal-value-line',
+  templateUrl: './modal-value-line.html'
+})
+export class ModalValueLineComponent extends AbstractModalLine {
+
+  @Input() set channelsToSubscribe(channels: ChannelAddress[]) {
+    this.channels = channels;
+  }
+
+  private channels: ChannelAddress[];
+
+  protected override getChannelAddresses(): ChannelAddress[] {
+    return this.channels;
+  }
+
+  @Input() private valueCallback: (currentData: CurrentData) => string;
```

Comment:
private here might create some compilation problems since the @Input() decorator is used. use public or default access modifiers.